StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSavedPaymentMethodsCollectionViewController.swift

```diff
@@ -537,6 +539,10 @@ extension CustomerSavedPaymentMethodsCollectionViewController: UpdateCardViewCon
 
         let updatedViewModel: Selection = .saved(paymentMethod: updatedPaymentMethod)
         viewModels[row] = updatedViewModel
+        // Update savedPaymentMethods
```

Comment:
This is unrelated to the default spm navigation but it's a fix for a bug I found— should I make a separate pr for that?

Reply:
Nice, can you explain the bug fix and add a test for it?

Reply:
Yeah, originally, there was a behavior I noticed in the horizontal flow where if the merchant country were FR and the currency were eur (leading to one CBC card and one non-editable visa card in the playground), if you change the CBC card brand and then remove the non-editable visa card, the card brand displayed for the CBC card will revert to what it was. I'm attaching a screen recording for reference— the screen recording was from before the changes we made in UI, but the behavior was always the same

Reply:
https://github.com/user-attachments/assets/5631b9d1-63a4-45dd-bcc7-e57f07237589

Reply:
This screen recording was before we decided on the changes, but the underlying behavior of the bug was the same. I can do a screen recording from master that shows the same.

Reply:
https://github.com/user-attachments/assets/f5f8b05d-99b1-4238-9cd4-1a1d6780413e